homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -60,13 +62,19 @@ async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
     except CannotConnect as ex:
         raise ConfigEntryNotReady from ex
     _LOGGER.debug("Devices: %s", devices)
-    coordinators: list[SwitchBotCoordinator] = []
+    coordinators_by_id: dict[str, SwitchBotCoordinator] = {}
     hass.data.setdefault(DOMAIN, {})
     data = SwitchbotCloudData(
         api=api,
         devices=SwitchbotDevices(
+            climates=[
+                prepare_device(hass, api, device, coordinators_by_id)
+                for device in devices
```

Comment:
We should only iterate the devices once and put them into the expected buckets. As this grows to support more devices we want to avoid a pattern where we iterate all the devices in a way that performs poorly because we iterate devices * device_types